mirgecom/initializers.py

```diff
@@ -262,6 +263,124 @@ def __call__(self, x_vec, *, t=0, eos=IdealSingleGas()):
                               momentum=mom)
 
 
+class DoubleMachReflection:
+    r"""Implement the double shock reflection problem.
+
+        - Woodward and Collela
+
+    The initial condition is defined
+
+    .. math::
+
+        {\rho}(x < x_s(y,t)) &= \gamma \rho_j\\
+        {\rho}(x > x_s(y,t)) &= \gamma\\
+        {\rho}{V_x}(x < x_s(y,t)) &= u_p \cos(\pi/6)\\
+        {\rho}{V_x}(x > x_s(y,t)) &= 0\\
+        {\rho}{V_y}(x > x_s(y,t)) &= u_p \sin(\pi/6)\\
+        {\rho}{V_y}(x > x_s(y,t)) &= 0\\
+        {\rho}E(x < x_s(y,t)) &= (\gamma-1)p_j\\
+        {\rho}E(x > x_s(y,t)) &= (\gamma-1)
+
+    where the shock position is given,
+
+    .. math::
+
+        x_s = x_0 + y/\sqrt{3} + 2 u_s t/\sqrt{3}
+
+    and the normal shock jump relations are
+
+    .. math::
+
+        \rho_j &= \frac{(\gamma + 1) u_s^2}{(\gamma-1) u_s^2 + 2} \\
+        p_j &= \frac{2 \gamma u_s^2 - (\gamma - 1)}{\gamma+1} \\
+        u_p &= 2 \frac{u_s^2-1}{(\gamma+1) u_s}
+
+    The initial shock location is given by $x_0$ and $u_s$ is the shock speed.
+    This function serves as an initial condition as well as boundary conditions
+    for $t>0$
```

Comment:
Actually, this object just creates the double mach solution only at time 0, and never anything else.  The argument (*t*) appears to be unused.  

Technically speaking, this routine is unrelated to the boundary condition. It simply creates the solution at locations *x_vec* - full stop.  IMO, I would remove anything having to do with "boundary condition" from the documentation, and also anything having to do with *t>0*, except maybe that *t* is unused.

Even if *t>0* was used and the routine could create the solution at *t>0*, then I still would document its ability to create a solution for *t>0*, but I'd leave the boundary conditions out of it.

Reply:
Resolved: https://github.com/illinois-ceesd/mirgecom/pull/213/commits/f3e6a6110968876e40be153abdb8fb5787b32ecd